### ISSUE TITLE
"Response Marshalling" does not ensure that required fields are present

### ISSUE THREAD

**Issue by JakeSummers** [noirbizarre/flask-restplus#478]
Good Morning,

I am attempting to use "response marshalling" to ensure that my api only returns valid data.  I was hoping that the "required" keyword would do this.  Unfortunately, this is not working for me.  I am curious if this is not a supported functionality, this is a bug or (most likely) I am doing something incorrectly.  

Code: 

```
from flask_restplus import fields
from routes.security import PublicResource
from restplus import api

ns = api.namespace('animals', description='Animal Stuff')

lul_cats_model = api.model('LulCats', {
    'cat_id': fields.String(description='The name or ID of the model',
                            required=True),
    'cat_name': fields.String(description='The version of the model',
                              required=False)
})


@ns.route('/lulcats')
class LulCats(PublicResource):
    @api.marshal_with(lul_cats_model)
    def get(self):
        return {
            'not_valid_field_name': 'fooBar'
        }
```

**Observed behaviour:** 
```
Response Code: 200
Response Body: 
{
  "cat_name": null,
  "cat_id": null
}
```

**Expected behaviour:** 

Not sure exactly.  I think that an exception should be thrown or a 500 should be returned.  The required fields are not present, so this should not return a 200.

---

Swagger file after hitting the `try it out` button: 

![image](https://user-images.githubusercontent.com/5657503/41663680-ffba562e-7471-11e8-9e1c-6262aa7d6c64.png)

**Comment by avilaton** [comment]
I think validation is ran only during parsing. You could run a quick test directly running http://flask-restplus.readthedocs.io/en/stable/api.html#flask_restplus.marshal on your schema and invalid data.

**Comment by JakeSummers** [comment]
Confirmed, marshal doesn't do validation:

```
from flask_restplus import fields, marshal
from restplus import api

lul_cats_model = api.model('LulCats', {
    'cat_id': fields.String(description='The name or ID of the model',
                            required=True),
    'cat_name': fields.String(description='The version of the model',
                              required=False)
})


data = {'cat_id':  'foo'}
out = marshal(data, lul_cats_model)
print('\nValid call to marshal - should (and does) work')
print(out)


data = {}
out = marshal(data, lul_cats_model)
print('\nInvalid call to marshal - shouldnt (but does) work')
print(out)
```

Outputs:

```

Valid call to marshal - should (and does) work
OrderedDict([('cat_id', 'foo'), ('cat_name', None)])

Invalid call to marshal - shouldnt (but does) work
OrderedDict([('cat_id', None), ('cat_name', None)])
```

Is it possible to flag this issue as a bug?

**Comment by ButtaKnife** [comment]
If you don't have that data and the fields are required, I'd expect the output to include those properties but as null, which is what happened.  It doesn't sound like a bug, but it sounds like you're requesting a new feature or a change in the existing behavior.

**Comment by JakeSummers** [comment]
@ButtaKnife - **What is the purpose of the `required` field if it doesn't enforce strong contracts?** 

With the current behaviour, the endpoint is allowed to return invalid results.  This kind of leak allows very hard to diagnose bugs.  If the endpoint is not able to return all of the required fields it should throw.  

Couple other thoughts: 

* It is often useful to log when you enter an invalid state.  This is automatic if `marshal` throws for invalid data. 
* APIs often have multiple clients.  If you need to validate that the API is adhering to it's contract in every client, this is a lot of extra code

Bottom Line: If you have a contract, you should adhere to it or throw an error

**Comment by strangecamelcaselogin** [comment]
Any progress on this issue?

**Comment by mircohaug** [comment]
Is there any way to strict validate the objects manually or during post_request handlers?

**Comment by fixedd** [comment]
[The documentation](https://flask-restplus.readthedocs.io/en/stable/marshalling.html#skip-fields-which-value-is-none) says to use `skip_none=True`, but that doesn't work either.

**Comment by j5awry** [comment]
Have you tried 

`@api.expect` rather than `@api.marshal_with`

you can use expect to validate

`@api.expect(model, validate=True)`

**Comment by ishtiaque-khan** [comment]
Any update on this?
